src/utils/internal/optionalRedirect.ts

```diff
@@ -17,19 +17,17 @@ export function optionalRedirect({
 
   const hasOnLoginRedirect = typeof onLoginRedirect === 'function';
 
-  const timeout = hasOnLoginRedirect ? 0 : DEFAULT_TIMEOUT;
-
   if (shouldRedirect && callbackRoute != null) {
-    setTimeout(() => {
```

Comment:
this is no longer pushing the execution in the event loop and might have side effects in some scenarios especially when dealing with browser window redirect which is a different thread from the normal JS execution thread